File: demo/src/pages/ImageDemo/ImageDemo.swift
//
//  ImageDemo.swift
//  playground
//
//  Created by Justin Spears on 9/13/22.
//

import SwiftUI

struct ImageDemo: View {
    var body: some View {
        VStack {
            Image(systemName: "rectangle.and.pencil.and.ellipsis")
            .imageScale(.small)
                .font(.system(size: 60))
            Image(systemName: "rectangle.and.pencil.and.ellipsis")
            .imageScale(.medium)
                .font(.system(size: 60))
            Image(systemName: "rectangle.and.pencil.and.ellipsis")
            .imageScale(.large)
                .font(.system(size: 60))
            }
    }
}

struct ImageDemo_Previews: PreviewProvider {
    static var previews: some View {
        ImageDemo()
    }
}

File: demo/src/pages/ImageDemo/ImageDemo.ts
import { Viewable, Image, Bound, VStack, Font, main } from "@tswift/ui";

@main
export class ImageDemo extends Viewable {
  body = VStack(
    Image("rectangle.and.pencil.and.ellipsis")
      .imageScale(".small")
      .font(Font.system(60)),
    Image("rectangle.and.pencil.and.ellipsis")
      .imageScale(".medium")
      .font(Font.system(60)),
    Image("rectangle.and.pencil.and.ellipsis")
      .imageScale(".large")
      .font(Font.system(60))
  );
}
